src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreManagerPackage.cs

```diff
@@ -114,5 +127,45 @@ private bool ShouldRestoreOnBuild
                 return packageRestoreConsent.IsAutomatic;
             }
         }
+
+        #region IVsUpdateSolutionEvents4
+
+        public void UpdateSolution_QueryDelayFirstUpdateAction(out int pfDelay)
+        {
+            // check if NuGet lock is already acquired by some other NuGet operation
+            if (LockService.IsLockHeld)
```

Comment:
Does `IsLockHeld` return true when *someone else* is holding the lock, or just you?